src/main/java/org/openpnp/gui/JobPanel.java

```diff
@@ -768,6 +810,10 @@ public boolean accept(File dir, String name) {
                 Job job = configuration.loadJob(file);
                 setJob(job);
                 addRecentJob(file);
+                
+                for (BoardLocation location : job.getBoardLocations()) {
```

Comment:
I believe this will cause the listener to be added every time refresh is called, and it will result in it being called once for every time it has been added, eventually slowing things down significantly. See https://docs.oracle.com/javase/7/docs/api/java/beans/PropertyChangeSupport.html#addPropertyChangeListener(java.beans.PropertyChangeListener).

Reply:
I make sure now to remove any listeners before adding the new ones. First thoughtthat would be a dirty way to do it but https://docs.oracle.com/javase/7/docs/api/java/beans/PropertyChangeSupport.html#addPropertyChangeListener(java.beans.PropertyChangeListener) says:
"If listener is null, or was never added, no exception is thrown and no action is taken." so that should not be a problem